core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -604,6 +604,21 @@ private[spark] class Executor(
           val serializedTK = ser.serialize(TaskKilled(killReason, accUpdates, accums, metricPeaks))
           execBackend.statusUpdate(taskId, TaskState.KILLED, serializedTK)
 
+        // When put the task in the pool, executor.stop may be called before task.run.
+        // The exception will be thrown from the task becauseof the unexpected status,
+        // see: SPARK-19147, here is to process the exception after executor.stop
+        // as the excepted exception.
+        case t: Throwable if !isLocal && env.isStopped =>
```

Comment:
CC @jerryshao or @squito perhaps
My question is, if the executor is shut down, can you even report metrics etc, or is it meaningful?

Reply:
you might succeed, as this will race against the stopping the executor.  But you're very likely to trigger more exceptions from `execBackend.statusUpdate`, so it probably doesn't make sense to try, especially if the whole point of this change is to cut down on scary error msgs during shutdown.

btw I think `env.isStopped` will need to be `volatile` for this to work reliably.

Reply:
@srowen @squito , thanks for the comments, I check the code again and make clearly how metrics & heartbeat work. You're right, report metrics is meaningless after executor.close(), because heartbeat won't work.
Update the pr and the exception will be processed in "case t: Throwable =>" part with log only.